plugin/tasks.py

```diff
@@ -35,103 +40,245 @@ def _decorator(*args, **kwargs):
-                              ctx.logger)
-    hostname = ctx.instance.runtime_properties['gcp_hostname']
-    response = gcp.delete_instance(hostname)
-    gcp.wait_for_operation(response['name'])
-    ctx.instance.runtime_properties.pop('gcp_hostname')
+def create_security_group(gcp_config, rules, **kwargs):
+    firewall = create_firewall_structure_from_rules(gcp_config['network'],
+                                                    rules)
+    firewall = FirewallRule(gcp_config,
+                            ctx.logger,
+                            firewall,
+                            gcp_config['network'])
+    firewall.create()
+    ctx.instance.runtime_properties[NAME] = firewall.name
 
 
 @operation
 @throw_cloudify_exceptions
-def create_network(config, **kwargs):
-    ctx.logger.info('Create network')
-    gcp = GoogleCloudPlatform(config['auth'],
-                              config['project'],
-                              config['scope'],
-                              ctx.logger)
-    network_name = utils.get_gcp_resource_name(config['network'])
-    response = gcp.create_network(network_name)
-    gcp.wait_for_operation(response['name'], True)
-    ctx.instance.runtime_properties['gcp_network'] = network_name
+def create_keypair(gcp_config,
+                   user,
+                   private_key_path,
+                   external,
+                   private_existing_key_path='',
+                   public_existing_key_path='',
+                   **kwargs):
+    keypair = KeyPair(gcp_config,
+                      ctx.logger,
+                      user,
+                      private_key_path)
+    if external:
+        keypair.private_key = ctx.get_resource(private_existing_key_path)
+        keypair.public_key = ctx.get_resource(public_existing_key_path)
+    else:
+        keypair.create()
+    keypair.add_project_ssh_key(user, keypair.public_key)
+    ctx.instance.runtime_properties['gcp_private_key'] = keypair.private_key
+    ctx.instance.runtime_properties['gcp_public_key'] = keypair.public_key
+    keypair.save_private_key()
 
 
 @operation
 @throw_cloudify_exceptions
-def delete_network(config, **kwargs):
-    ctx.logger.info('Delete network')
-    gcp = GoogleCloudPlatform(config['auth'],
-                              config['project'],
-                              config['scope'],
-                              ctx.logger)
-    network_name = ctx.instance.runtime_properties['gcp_network']
-    response = gcp.delete_network(network_name)
-    gcp.wait_for_operation(response['name'], True)
-    ctx.instance.runtime_properties.pop('gcp_network')
+def save_private_key(gcp_config,
+                     private_key,
+                     user,
+                     private_key_path,
+                     **kwargs):
+    keypair = KeyPair(gcp_config,
+                      ctx.logger,
+                      user,
+                      private_key_path)
+    keypair.private_key = private_key
```

Comment:
I had to change some things to get boostrap working with this branch. I'll investigate it.